src/libopensc/card-oberthur.c

```diff
@@ -1345,12 +1346,16 @@ auth_update_component(struct sc_card *card, struct auth_update_component_info *a
 		    LOG_FUNC_RETURN(card->ctx, SC_ERROR_OUT_OF_MEMORY);
 
 		p2 = 0;
-		if (args->len == 24)
-			EVP_EncryptInit_ex(ctx, EVP_des_ede(), NULL, args->data, NULL);
-		else
-			EVP_EncryptInit_ex(ctx, EVP_des_ecb(), NULL, args->data, NULL);
+		if (args->len == 24) {
+			alg = sc_evp_cipher(card->ctx, "DES-EDE");
+			EVP_EncryptInit_ex(ctx, alg, NULL, args->data, NULL);
+		} else {
+			alg = sc_evp_cipher(card->ctx, "DES-ECB");
+			EVP_EncryptInit_ex(ctx, alg, NULL, args->data, NULL);
```

Comment:
nit: the second line in both branches is the same so I would prefer to keep it out of the branching once we already touch this.

Reply:
ok